lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -323,6 +323,12 @@ static void createMergeArgReady(ArrayRef<Value> outputs, Value fired,
   }
 }
 
+static void extractValues(ArrayRef<ValueVector *> valueVectors, size_t index,
+                          SmallVectorImpl<Value> &result) {
+  for (size_t i = 0; i < valueVectors.size(); ++i)
+    result.push_back((*valueVectors[i])[index]);
+}
```

Comment:
I'd recommend using a foreach loop:

  for (auto elt: valueVectors)
    result.push_back((*elt)[index]);